### PR TITLE
added update submodule before api generation

### PR DESCRIPTION
**Pull request by rigazilla**
Fixes gingersnap-project/api#9

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,12 +111,16 @@ API_GO_FILES = api/v1alpha1/zz_cache.pb.go api/v1alpha1/zz_rules.pb.go
...
+update-git-submodules-remote: ## Pull the latest changes from remote upstream HEAD
 	git submodule update --init --remote
 
+.PHONY: update-git-submodules
+update-git-submodules: ## Sync submodule content with parent project checkout
+	git submodule update --init
+
 .PHONY: gingersnap-api-generate
-gingersnap-api-generate: protoc protoc-gen-go protoc-gen-deepcopy $(API_GO_FILES) ## Generate code for gingersnap-api
+gingersnap-api-generate: update-git-submodules protoc protoc-gen-go protoc-gen-deepcopy $(API_GO_FILES) ## Generate code for gingersnap-api
```
Comment (ryanemerson) [inline Makefile:123]: IMO we shouldn't add `update-git-submodules`  to the generate target. What if I want to pull from remote then generate?

If we omit `update-git-submodules` on `gingersnap-api-generate` we can do either:

`make update-git-submodules-remote  generate` or `make update-git-submodules generate`

Comment (rigazilla) [reply Makefile:123]: `update-git-submodules` only ensure that the content of gingersnap-api folder is aligned with the commit "pointed" by the parent project. I think it's safe to check it before each build.

`update-git-submodules-remote` does something different: it updates the parent project to point to the HEAD of the submodule branch and then updates the gingersnap-api folder.
So basically if you run a remote update, the following executions of `update-git-submodules` just ensure that the folder is aligned with the remote HEAD.

well let me check if all this is true :)

Comment (ryanemerson) [reply Makefile:123]: Thanks for confirming

**Comment by ryanemerson** [comment]
Thanks @rigazilla

**Pull request merged**